storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -95,13 +108,86 @@ func (s *commitLogSource) Read(
 	defer iter.Close()
 
 	var (
-		unmerged    = make(map[uint32]map[ts.Hash]encoderMap)
+		numShards = len(shardsTimeRanges)
+		unmerged  = make([]map[ts.Hash]encoderMap, numShards, numShards)
```

Comment:
Revert to `make(map[uint32]map[ts.Hash]encoderMap)` instead of `[]map[ts.Hash]encoderMap`: shard allocations aren't contiguous or zero-based, so a node could have `numShards=10` and the shards could be `1,10,20,30,40..`

You could pre-allocate the map to have size `numShards` if that helps perf